build/WFCORE-3769_WFLY-10305_galleon_build.adoc

```diff
@@ -0,0 +1,166 @@
+   org.wildfly.build:wildfly-server-provisioning-maven-plugin. These will use
+   the 'legacy core feature pack' discussed previously.
+ ** This maven profile should not run by default.
+ * The WildFly maven build should include a profile that builds
+   new maven modules that produce 'legacy servlet and full thin and fat distributions' 
+   using the legacy org.wildfly.build:wildfly-server-provisioning-maven-plugin. These will
+   use the 'legacy servlet feature pack' and 'legacy full feature pack' discussed previously.
+ ** This maven profile should not run by default.
+ * These legacy distributions should be produced in such a way that their output, except 
+   for their maven GAVs, would be acceptable for release as the primary build output if 
+   this shift to Galleon had not occurred. In other words, these distributions should 
+   provide a valid base for comparison to the output of the Galleon build.
+ ** The only purpose of these legacy distributions is to facilitate such comparisons.
+ * The content of the Galleon-produced distributions should functionally match the output
+   of the legacy distributions, with only minimal minor cosmetic differences.
+ * Special distributions of the server that are produced inside the WildFly Core and 
+   WildFly testsuites should be produced by Galleon, not by legacy tooling.
+ ** Copying of distribution content produced in other modules of the build into the
+    needed testsuite locations is acceptable; i.e. there is no requirement that
+    all test distributions be built using Galleon, just as there was no requirement
+    in the past to use the legacy server-provisioning plugin in all places.
+
+=== Nice-to-Have Requirements
+
+None 
+
+=== Non-Requirements
+
+ * Support for any use of the Galleon tooling outside of the maven build steps described in this document. 
+ * Long term support for the initial content details of the Galleon feature packs. The initial version
+   of these feature packs should be regarded as 'Tech Preview', i.e. subject to change in the next WildFly 
+   release if we discover the need to correct things after this initial use. The intent though is to produce 
+   feature packs external parties can look at to learn about how WildFly+Galleon will work and to
+   start planning how their own software can be based on our feature packs. 
+ * Publication of legacy 'thin' and 'fat' distributions to public maven repositories. These are
+   only meant for testing use to verify the Galleon output against the current mechanism output;
+   they are not meant for other consumption. Publishing these is actually an anti-requirement.
+ * Absolutely identical output content of a thin or fat distribution produced using the legacy tooling and 
+   one produced using Galleon. The two are meant to be very, very similar but minor differences that do
+   not affect functionality are acceptable. The intent though is to minimize these.
```

Comment:
`differences that do not affect functionality are acceptable` - what fits to this category ?

Some real examples would be nice. The tool has PR to be merged to WildFly Core, PR for WildFLy should come soon. We should be aware of existing minor differences.

Reply:
Like the ones I mentioned before. There is for example a new .galleon dir which records the provisioned state. This directory could be removed actually assuming nobody will try reprovisioning the installation using galleon.